webapp/sources/utils/src/main/scala/com/normation/ZioCommons.scala

```diff
@@ -108,6 +111,29 @@ object errors {
     def attemptZIO[A](ioeffect: => IOResult[A]):                IOResult[A]        = {
       attemptZIO("An error occurred")(ioeffect)
     }
+
+    // for non-blocking case
+    def nonBlocking[A](error: String)(effect: => A): IO[SystemError, A] = {
+      // in ZIO 2.1, attempt is for non blocking, CPU bound effects
+      ZIO.attempt(effect).mapError(ex => SystemError(error, ex))
+    }
```

Comment:
Do we want to migrate all non blocking effects to this use this name ?
If so, I find the name very specific to knowing the zio internals, I would expect a more standard nomenclature such as "lift"/"attemptSucceed" when looking for such a method, and we would have many places with `nonBlocking()` semantics in our effectful code without I/O e.g. for error-handling

Reply:
For latter reference: 
- the goal is to have an as clean as possible split between blocking and non blocking, 
- in doubt, the default is blocking because the platform is not operated by us, and we risk a deadlock with blocking effects in the non blocking pool, which we don't want to impose to users. 
- nonBlocking is almost only for encapsulating non blocking exceptions (like parsing exception, int conversion, etc) - but then, these one, we prefer to modelize them with `PureResult`
- blocking effects are: 
    - anything that leads the underlying thread to go into BLOCK, WAIT, TIMED WAIT (but not limited to, because a RUNNING thread can be waiting for a syscal),
    - anything leading to a blocking syscall (need to read Linux/Windows API doc for that), 
        - in particular any I/O, including but not limited to sockets, files (including writing to console or logs), etc
    - any blocking delimiters like mutex etc (but normally, that leads the underlying thread to WAIT or BLOCK)